File: examples/android/CMeSdkJava/app/src/main/java/me/connect/sdk/java/sample/db/dao/StructuredMessageDao.java
package me.connect.sdk.java.sample.db.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import me.connect.sdk.java.sample.db.entity.StructuredMessage;

@Dao
public interface StructuredMessageDao {

    @Query("SELECT * FROM structuredmessage")
    LiveData<List<StructuredMessage>> getAll();

    @Query("SELECT * FROM structuredmessage WHERE id = :id")
    StructuredMessage getById(int id);

    @Insert
    void insertAll(StructuredMessage... messages);

    @Query("SELECT EXISTS(SELECT * FROM structuredmessage WHERE (entry_id = :entryId AND pwdid = :pwDid))")
    boolean checkMessageExists(String entryId, String pwDid);

    @Query("SELECT * FROM structuredmessage WHERE (entry_id = :entryId AND pwdid = :pwDid)")
    StructuredMessage getByEntryIdAndPwDid(String entryId, String pwDid);

    @Update
    void update(StructuredMessage message);

    @Delete
    void delete(StructuredMessage message);
}

File: examples/android/CMeSdkJava/app_kt/src/main/kotlin/me/connect/sdk/java/samplekt/db/dao/StructuredMessageDao.kt
package me.connect.sdk.java.samplekt.db.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import me.connect.sdk.java.samplekt.db.entity.StructuredMessage


@Dao
interface StructuredMessageDao {
    @Query("SELECT * FROM structuredmessage")
    fun getAll(): LiveData<List<StructuredMessage>>

    @Query("SELECT * FROM structuredmessage WHERE id = :id")
    suspend fun getById(id: Int): StructuredMessage

    @Insert
    suspend fun insertAll(vararg messages: StructuredMessage)

    @Query("SELECT EXISTS(SELECT * FROM structuredmessage WHERE pwDid = :pwDid)")
    fun checkMessageExists(pwDid: String): Boolean

    @Update
    suspend fun update(message: StructuredMessage)

    @Delete
    suspend fun delete(message: StructuredMessage)
}
